packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx

```diff
@@ -15,12 +16,18 @@ export const TableRowMenuProvider = ({ value, children }: TableRowMenuProviderPr
 
   const onMouseOverRow = useCallback(
     (rowRef: React.RefObject<HTMLDivElement>) => {
-      if (isMenuOpen || hoveredRowRef?.current === rowRef.current) return;
+      if (hasMenuRef.current === false || isMenuOpen || hoveredRowRef?.current === rowRef.current) return;
+
       resetTimeout();
       setHoveredRowRef(rowRef);
       const tableRootTop = tableRootRef.current.getBoundingClientRect().top;
       const rowTop = rowRef.current.getBoundingClientRect().top;
       setMenuButtonPosition(rowTop - tableRootTop);
+      if (hasMenuRef.current === null) {
```

Comment:
|| true?

Reply:
or even
```suggestion
      if (hasMenuRef.current !== false) {
```

Reply:
to avoid re-entering this if

Reply:
Only the first time it is null. After that it is set to true/false. So checking if it is null will run only the first time. If I check not false than the check will run every time for the tables that don't have a menu

Reply:
ohhh you're right actually
sorry :)